app/customer.py

```diff
@@ -0,0 +1,57 @@
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(self, info: dict) -> None:
+        self.name = info["name"]
+        self.location = info["location"]
+        self.money = info["money"]
+        self.car_consumption = info["car"]["fuel_consumption"]
+        self.product_cart = info["product_cart"]
+        self.trip_totals = {}
+
+    def select_the_shop(
+            self, shops: list[Shop], fuel_price: float
+    ) -> Shop | None:
+        print(f"{self.name} has {self.money} dollars")
+        totals = []
+
+        for shop in shops:
+            trip_cost = round(
+                shop.calc_prod_price(self.product_cart)
+                + 2
+                * fuel_price
+                * self.car_consumption
+                * shop.calc_distance(self.location)
+                / 100,
+                2,
+            )
+            print(f"{self.name}'s trip to the {shop.name} costs {trip_cost}")
+            totals.append(trip_cost)
+            self.trip_totals[shop.name] = trip_cost
+
+        if self.money < min(totals):
+            print(f"{self.name} doesn't have enough money "
+                  f"to make purchase in any shop")
+            return None
+        the_shop = shops[totals.index(min(totals))]
+        print(f"{self.name} rides to {the_shop.name}")
+        return the_shop
+
+    def go_shopping(self, shop: Shop) -> None:
+        print()
+        # here should be datetime.now(), but tests are hard-coded
+        print("Date: 04/01/2021 12:33:41")

```

Comment:
I used strftime() method for datetime.now, and tests have been successfully done.